docs/design/proposals/metro-dr.md

```diff
@@ -0,0 +1,216 @@
+    * Possibly derived using kustomize during initial deployment to a chosen
+      affined StorageClass for the cluster
+
+* The pool name should indicate the primary availability zone.
+* Special Crushrules on the pool to create primary affinity for it. This means
+  that the primary OSD will be chosen from the OSDs on the same availability
+  zone as the kubernetes cluster.
+* In each kubernetes cluster we will have an cephcsi that connects to the
+  external ceph cluster
+  * storageclass per pool (per availability zone)
+  * (optional) A default storageclass using the pool matching the cluster
+    availability zone, this will require special handling failover.
+  * All PVC for specific availability zone that use the default storage cluster
+    will have affinity
+
+For CephFS the metadata pool will be shared across the availability zone. As a
+future enhancement we can use MDS subtree pinning to improve metadata
+operations performance.
+
+**Scenario 3: Adding a peer cluster to an existing kubernetes cluster with
+existing provisioned PVCs**
+
+* NOTE: As existing ceph-csi would already have used the empty `uniqueID`, only
+  one such cluster can be added as a peer to reuse existing provisioned PVCs.
+  Adding more than one such peer would result in more than one of them using
+  the empty `uniqueID` and will result in deployments not getting their storage
+  bound when migrated to peer clusters.
+
+* Kubernetes cluster-a
+  * ceph-csi config map
+    * clusterID: ceph-cluster-foo
+    * NOTE: no `uniqueID` as cluster was setup prior to DR needs
+* Kubernetes cluster-b
+  * ceph-csi config map
+    * clusterID: ceph-cluster-foo
+      * uniqueID: baz
+* StorageClass-a:
+  * clusterID: ceph-cluster-foo
+  * PVCNaming: true
+
+**NOTE**: The above parameter should be set prior to provisioning volumes that
+may in the future be migrated to peer kubernetes clusters, as it is used as the
+identifying signature to find backend images for the request PVC:
+
+* PVC:
+  * StorageClassName: StorageClass-a*
+
+### For snapshot
+
+This will be the same as the CreateVolume request, here instead of the
+storageclass parameter, cephcsi will check the snapshotclass parameter.
+
+Same as PVC name and namespace name to generate the volume name. We will use
+snapshot name and snapshot namespace name to generate the snapshot name,
+created an
+[issue](https://github.com/kubernetes-csi/external-snapshotter/issues/341) in
+external snapshotter to pass extra snapshot metadata to create the snapshot
+request.
+
+### For Encrypted PVC
+
+Currently, volumeID is used as the key to store the encryption key in KMS, the
+encryption will not work in case the workload is moved to another cluster (the
+voluemID will be completely different than the volumeID on the previous
+cluster). The common parameter across the cluster is the volume name, So
+cephcsi need to store the volume name instead of volumeID in KMS.
```

Comment:
... or store the `key` (to the encryption key) in the image's metadata.

Reply:
sounds good for rbd. can this be done for cephfs also @ShyamsundarR wihtout mounting the cephfs subvolumes?

Reply:
I'd imagine you can do something similar using a flat file co-located with the subvolumes.